standalone-metastore/metastore-server/src/main/resources/package.jdo

```diff
@@ -1655,6 +1655,49 @@
           </value>
        </field>
     </class>
+    <class name="MDCPrivilege" table="DC_PRIVS" identity-type="datastore" detachable="true">
+          <index name="DCPrivilegeIndex" unique="true">
+            <column name="AUTHORIZER"/>
+            <column name="DC_NAME"/>
+            <column name="PRINCIPAL_NAME"/>
+            <column name="PRINCIPAL_TYPE"/>
+            <column name="DC_PRIV"/>
+            <column name="GRANTOR"/>
+            <column name="GRANTOR_TYPE"/>
+          </index>
+
+          <datastore-identity>
+            <column name="DC_GRANT_ID"/>
+          </datastore-identity>
+
+          <field name="principalName">
+            <column name="PRINCIPAL_NAME" length="128" jdbc-type="VARCHAR"/>
+          </field>
+          <field name="principalType">
+            <column name="PRINCIPAL_TYPE" length="128" jdbc-type="VARCHAR"/>
+          </field>
+          <field name="dataConnector">
```

Comment:
so MDCPrivilege.dataConnector is of type MDataConnector whereas this mapping indicates it is a string field. Shouldnt these types match? Am I missing something here?

Reply:
This is intended. MDCPrivilege has dataconnector as MDataConnector, but dataconnector itself is actually stored as String in backend Table (because it cannot store dataconnector type). 

The reason why MDCPrivilege can retrieve dataconnector as MDataConnector even though it is stored as String in backendTable is that DC_NAME here actually serving as a Foreign Key referencing NAME in DATACONNECTORS table(as specified in schema when init).